pontoon/sync/tasks.py

```diff
@@ -238,14 +240,17 @@ def sync_translations(
         repos = repos.filter(pk__in=repo_locales.keys())
         log.info('Pulling changes for project {0} complete.'.format(db_project.slug))
 
-    resources_changed = []
+    changed_resources = None
     obsolete_vcs_entities = []
+
     if project_changes:
-        resources_changed = (
-            project_changes['update_db'] +
-            project_changes['obsolete_db'] +
-            project_changes['create_db']
+        updated_entity_pks = [x[1].pk for x in project_changes['update_db']]
+        obsolete_entity_pks = project_changes['obsolete_db']
+        changed_resources = db_project.resources.filter(
+            Q(entities__date_created=now) |
+            Q(entities__pk__in=updated_entity_pks + obsolete_entity_pks)
         )
```

Comment:
* `project_changes['update_db']`: entities to update in DB, if e.g. string or comment changes and key doesn't
* `project_changes['obsolete_db']`: entities to obsolete
* `project_changes['create_db']`: entities to create in DB

Since `project_changes['create_db']` doesn't contain PKs, we need to fetch those entities using `Q(entities__date_created=now)`. All newly created entities have the same timestamp as the sync (`now`).

Alternative approach would be to gather `changed_resources` while syncing sources and pass them all the way here, but I'd rather do this as a followup to this PR (to avoid making it even bigger).

Reply:
The 'now' query has a race condition with regular translations, and other sync process, right?

Do we use bulk_insert? https://docs.djangoproject.com/en/1.11/ref/models/querysets/#bulk-create says that starting with 1.10, on postgres, we get the IDs back.

Reply:
> The 'now' query has a race condition with regular translations, and other sync process, right?

No. Regular translations are stored in a different model (Translation) than source strings (Entity). Other sync processes for the same project [cannot happen at the same time](https://github.com/mozilla/pontoon/blob/master/pontoon/sync/core.py#L37).

> Do we use bulk_insert? https://docs.djangoproject.com/en/1.11/ref/models/querysets/#bulk-create says that starting with 1.10, on postgres, we get the IDs back.

No, but we [should](https://github.com/mozilla/pontoon/blob/master/pontoon/sync/changeset.py#L167). Thanks. Although that wouldn't bring us IDs here.

Reply:
'k